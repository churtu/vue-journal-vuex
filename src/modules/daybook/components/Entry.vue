<template>
    <div class="mb-3 pointer p-2 entry-container" @click="$router.push({name:'entry', params:{id:10}})">
        <div class="entry-title d-flex">
            <span class="text-success fs-5 fw-bold">29</span>
            <span class="mx-1 fs-5">Enero</span>
            <span class="mx-2 fw-light">2022, Sabado</span>
        </div>
        <div class="entry-description">
            Lorem nulla deserunt laborum laboris ut irure occaecat sunt aliqua nostrud ad reprehenderit. Sint culpa cupidatat cupidatat occaecat ea cupidatat amet aute aliqua laborum do eu. Labore ex aliqua Lorem velit.
        </div>
    </div>
</template>

<style lang="scss" scoped>
    .entry-container{
        border-bottom: solid 1px #2c3e50;
        transition: .2s all ease-in;

        &:hover{
            background: lighten($color: grey, $amount: 45);
        }

        .entry-description{
            font-size: 12px;
        }
    }
</style>