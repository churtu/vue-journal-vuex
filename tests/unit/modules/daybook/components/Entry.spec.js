import { shallowMount } from '@vue/test-utils'
import { journalState } from '../../../mock-data/test-journal-state'

import Entry from '@/modules/daybook/components/Entry'

describe('Pruebas en el Entry component', () => {
    //mock router
    const mockRouter = {
        push: jest.fn()
    }
    // wrapper, con props y global para configurar el router
    const wrapper = shallowMount(Entry, {
        props:{
            entry: journalState.entries[0]
        },
        global: {
            mocks: {
                $router: mockRouter
            }
        }
    })

    test('Debe hacer match con el Snapshot', () => { 
        expect(wrapper.html()).toMatchSnapshot()
     })

    test('Debe de direccionar al hacer click en el .entry-container', () => { 
        wrapper.find('.entry-container').trigger('click')
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: journalState.entries[0].id
            }
        })
     })

    test('Pruebas en las propiedades computadas', () => {
        expect( typeof wrapper.vm.day ).toBe('number')
        expect( typeof wrapper.vm.month ).toBe('string')
        expect( typeof wrapper.vm.yearDate ).toBe('string')
    })
})