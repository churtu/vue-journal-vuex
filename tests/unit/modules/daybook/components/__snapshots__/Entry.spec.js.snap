// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry component Debe hacer match con el Snapshot 1`] = `
<div class="mb-3 pointer p-2 entry-container">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">2</span><span class="mx-1 fs-5">Febrero</span><span class="mx-2 fw-light">2022, Mi√©rcoles</span></div>
  <div class="entry-description">Soy un mock</div>
</div>
`;
